dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/auth/AuthScheme.java

```diff
@@ -50,16 +51,23 @@
     include = JsonTypeInfo.As.EXISTING_PROPERTY,
     property = "type")
 @JsonSubTypes({
-  @JsonSubTypes.Type(value = HttpBasicAuth.class, name = "http-basic"),
-  @JsonSubTypes.Type(value = ApiTokenAuth.class, name = "api-token")
+  @JsonSubTypes.Type(value = HttpBasicAuthScheme.class, name = "http-basic"),
+  @JsonSubTypes.Type(value = ApiTokenAuthScheme.class, name = "api-token"),
+  @JsonSubTypes.Type(value = ApiHeadersAuthScheme.class, name = "api-headers"),
+  @JsonSubTypes.Type(value = ApiQueryParamsAuthScheme.class, name = "api-query-params")
 })
-public abstract class Auth implements Serializable {
+public abstract class AuthScheme implements Serializable {
```

Comment:
I don't know much about this domain but using sub-classing seems overkill and unnecessarily complex to me. Generally with data I would recommend to stay clear of class hierarchies and instead try to find a common way of representing the information. For example, adding a `Map<String,String>` to keep the data and an enum type to indicate which of the auth types the data represents might work. Just a suggestion.